de.fu_berlin.inf.dpp.core/src/de/fu_berlin/inf/dpp/negotiation/InstantOutgoingProjectNegotiation.java

```diff
@@ -0,0 +1,273 @@
+        PipedInputStream in = new PipedInputStream();
+        OutputStream out = new PipedOutputStream(in);
+
+        String userID = getPeer().toString();
+        OutgoingFileTransfer transfer;
+        transfer = fileTransferManager.createOutgoingFileTransfer(userID);
+
+        try {
+            OutgoingStreamProtocol osp;
+            osp = new OutgoingStreamProtocol(out, session, monitor);
+
+            /* id in description needed to bypass SendFileAction handler */
+            String streamName = TRANSFER_ID + getID();
+            transfer.sendStream(in, streamName, 0, streamName);
+
+            awaitNegotation(transfer, monitor);
+
+            sendProjectConfigFiles(osp);
+            sendRemainingPreferOpenedFirst(osp);
+
+            osp.close();
+        } finally {
+            IOUtils.closeQuietly(out);
+        }
+
+        monitor.done();
+        log.debug(this + ": file transfer done");
+    }
+
+    @Override
+    protected void cleanup(IProgressMonitor monitor) {
+        editorManager.removeSharedEditorListener(listener);
+        session.userStartedQueuing(remoteUser);
+
+        if (stoppedUsers != null)
+            startUsers(stoppedUsers);
+
+        super.cleanup(monitor);
+    }
+
+    private void createTransferList(List<FileList> fileLists, int fileCount) {
+        List<SPath> files = new ArrayList<SPath>(fileCount);
+        for (final FileList list : fileLists) {
+            IProject project = session.getProject(list.getProjectID());
+            for (String file : list.getPaths()) {
+                files.add(new SPath(project.getFile(file)));
+            }
+        }
+
+        /* sort hierarchy based, top files are seen first in project explorer */
+        Collections.sort(files, new Comparator<SPath>() {
+            @Override
+            public int compare(SPath a, SPath b) {
+                int lenA = a.getProjectRelativePath().segmentCount();
+                int lenB = b.getProjectRelativePath().segmentCount();
+                return Integer.valueOf(lenA).compareTo(Integer.valueOf(lenB));
+            }
+        });
+
+        /* LinkedHashSet for fast lookup while keeping sort order */
+        transferList = new LinkedHashSet<SPath>(files);
+    }
+
+    private void fileOpened(SPath file) {
+        if (file != null) {
+            openedFiles.addFirst(file);
+            log.debug(this + ": added " + file + " to open files queue");
+        }
+    }
+
+    private void sendProjectConfigFiles(OutgoingStreamProtocol osp)
+        throws IOException, LocalCancellationException {
+        /* TODO should be configurable in future */
+        String[] eclipseProjFiles = {
+            ".settings/org.eclipse.core.resources.prefs" /* for file encoding! */,
+            ".classpath", ".project",
+            ".settings/org.eclipse.core.runtime.prefs",
+            ".settings/org.eclipse.jdt.core.prefs",
+            ".settings/org.eclipse.jdt.ui.prefs" };
```

Comment:
I am not sure this is a good idea to include in the CORE.
These are highly eclipse specific files and although that is the main target, this does not belong here.

You should instead provide an interface to add config files for the ide specific plugins, but not list any here by default.

Reply:
updated #139 with this info